File: ios/App/GraphQL/Typedefs.swift
import AppGraphqlApi
import Foundation

// Keep generated GraphQL selection types behind local aliases so feature code
// does not depend directly on generated module paths.
typealias CurrentUserData = GetCurrentUserQuery.Data.CurrentUser
typealias UpdatedUserData = UpdateUserMutation.Data.UpdateUser
typealias UsersConnectionData = GetUsersQuery.Data.Users
typealias UserRowData = GetUsersQuery.Data.Users.Node
typealias ProjectsConnectionData = GetProjectsQuery.Data.Projects
typealias ProjectRowData = GetProjectsQuery.Data.Projects.Node
typealias CreatedProjectData = CreateProjectMutation.Data.CreateProject
typealias UpdatedProjectData = UpdateProjectMutation.Data.UpdateProject
typealias AiVoiceSessionData = CreateAiVoiceSessionMutation.Data.CreateAiVoiceSession
typealias PaymentSubscriptionData = MySubscriptionQuery.Data.MySubscription
typealias BillingPortalSessionData = CreateBillingPortalSessionMutation.Data.CreateBillingPortalSession
typealias SubscriptionCheckoutSessionData = CreateSubscriptionCheckoutSessionMutation.Data.CreateSubscriptionCheckoutSession
typealias UploadSlotData = CreateUploadMutation.Data.CreateUpload
typealias FinalizedUploadData = FinalizeUploadMutation.Data.FinalizeUpload
typealias PushDeviceData = RegisterPushDeviceMutation.Data.RegisterPushDevice
typealias PushDevicePlatformValue = AppGraphqlApi.PushDevicePlatform
typealias ProjectStatusValue = AppGraphqlApi.ProjectStatus
typealias UserStatusValue = AppGraphqlApi.UserStatus
typealias SubscriptionStatusValue = AppGraphqlApi.SubscriptionStatus
typealias SubscriptionIntervalValue = AppGraphqlApi.SubscriptionInterval

extension CurrentUserData {
  var stringId: String { String(id) }
}

File: android/app/src/main/kotlin/com/baseapp/android/graphql/Typedefs.kt
package com.baseapp.android.graphql

// Keep generated GraphQL types behind local aliases so feature code does not
// depend directly on generated package paths. Rows are the shared fragment
// models (Apollo Kotlin's operationBased codegen exposes fragments as their
// own data classes).
typealias CurrentUserData = com.baseapp.android.graphql.generated.fragment.UserFields
typealias UpdatedUserData = com.baseapp.android.graphql.generated.fragment.UserFields
typealias UserRowData = com.baseapp.android.graphql.generated.fragment.UserFields
typealias ProjectRowData = com.baseapp.android.graphql.generated.fragment.ProjectFields
typealias UsersConnectionData = com.baseapp.android.graphql.generated.GetUsersQuery.Users
typealias ProjectsConnectionData = com.baseapp.android.graphql.generated.GetProjectsQuery.Projects
typealias CreatedProjectData = com.baseapp.android.graphql.generated.fragment.ProjectFields
typealias UpdatedProjectData = com.baseapp.android.graphql.generated.fragment.ProjectFields
typealias ProjectStatusValue = com.baseapp.android.graphql.generated.type.ProjectStatus
typealias UserStatusValue = com.baseapp.android.graphql.generated.type.UserStatus
typealias SortDirectionValue = com.baseapp.android.graphql.generated.type.SortDirection

// Payments kernel (Settings Billing card + subscribe flow twins).
typealias PaymentSubscriptionData = com.baseapp.android.graphql.generated.fragment.PaymentSubscriptionFields
typealias CheckoutSessionData = com.baseapp.android.graphql.generated.fragment.CheckoutSessionFields
typealias SubscriptionStatusValue = com.baseapp.android.graphql.generated.type.SubscriptionStatus
typealias SubscriptionIntervalValue = com.baseapp.android.graphql.generated.type.SubscriptionInterval

// Storage kernel (avatar upload twin).
typealias UploadSlotData = com.baseapp.android.graphql.generated.CreateUploadMutation.CreateUpload
typealias UploadVisibilityValue = com.baseapp.android.graphql.generated.type.UploadVisibility

// Push kernel (Settings Notifications card twin).
typealias PushDeviceData = com.baseapp.android.graphql.generated.RegisterPushDeviceMutation.RegisterPushDevice
typealias PushDevicePlatformValue = com.baseapp.android.graphql.generated.type.PushDevicePlatform
